### DISCUSSION TITLE
Tried to delete a flow but failed

### DISCUSSION THREAD

**Discussion by BrunoUY** [directus/directus#14463] [Archive]
I tried to delete a flow but failed, also I tried to delete the connections to the Event hook but the same error appears.

![imagen](https://user-images.githubusercontent.com/4197644/179292046-f22e5a68-1b1b-4088-9f09-89f169287ab8.png)

```
{
  "errors": [
    {
      "message": "delete from `directus_flows` where `id` in ('a8cef0e4-0c4c-4a50-87c0-c1bcc4e1ad86') - ER_ROW_IS_REFERENCED_2: Cannot delete or update a parent row: a foreign key constraint fails (`asset_screener_v2`.`directus_operations`, CONSTRAINT `directus_operations_reject_foreign` FOREIGN KEY (`reject`) REFERENCES `directus_operations` (`id`))",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

Now, **I Can´t create content** because the flow was altering the items.create and items.update
( I set it to inactive and it works, but not delete operation on the flow )

I'm running in a local environment.

any idea on this?

thank you, Bruno.

**Comment by azrikahar** [comment]
Are you using MySQL/Mariadb? 

I tried to construct a scenario based on the hint ``FOREIGN KEY (`reject`)``, and it seems like it might be due to a form of race condition where the operations are not deleted in the "right" order:

https://user-images.githubusercontent.com/42867097/179479178-60e9e487-726b-4b62-99d2-f8e5eb910686.mp4

Can you share the exact flow setup you have? Just so we can have a reference in case we'd need to reproduce this. 

After that, can you try what I did, which is to remove some of the `reject` operations and try to delete the flow again? May require more removal of reject operations depending on how many chains your flow has.
